pandas/core/frame.py

```diff
@@ -390,8 +390,9 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
             else:
                 mgr = self._init_ndarray(data, index, columns, dtype=dtype,
                                          copy=copy)
-        elif isinstance(data, (list, types.GeneratorType)):
-            if isinstance(data, types.GeneratorType):
+        elif (isinstance(data, collections.Iterable)
```

Comment:
can you add a comment here (1 line above) which defines what this clause is checking